command.go

```diff
@@ -153,6 +153,35 @@ func (c *Command) RunInDirPipelineWithTimeout(timeout time.Duration, stdout, std
 	case err = <-result:
 		return err
 	}
+
+}
+
+// RunInDirPipelineWithTimeout executes the command in given directory and
+// timeout duration. It pipes stdout and stderr to supplied io.Writer.
+// DefaultTimeout will be used if the timeout duration is less than
+// time.Nanosecond (i.e. less than or equal to 0). It returns an ErrExecTimeout
+// if the execution was timed out.
+func (c *Command) RunInDirPipelineWithTimeout(timeout time.Duration, stdout, stderr io.Writer, dir string) (err error) {
```

Comment:
I find this method name super confusing with `RunInDirWithPipelineTimeout`, and the diff in their names does not actually tell the different behavior of two methods.

Also 

```go
	if logOutput != nil {
		buf.Grow(512)
		w = &limitDualWriter{
			W: buf,
			N: int64(buf.Cap()),
			w: stdout,
		}
	}
```

is only limiting the log output, not the output to the stdout, so I think we'd want this either case.

I propose the new method name to be `RunInDirWithOptions(dir string, opts RunInDirOptions)`, where `RunInDirOptions` contains timeout, stdin, stdout, stderr.